### PR TITLE
Change Command keycode defaults to match corresponding Bootmagic codes

### PR DESCRIPTION
**Pull request by vomindoraan**
**tl;dr:** Make Command keys more intuitive/familiar so hopefully more people will use Command.

## Description
Currently, the [Command](https://docs.qmk.fm/#/feature_command?id=configuration) feature isn't used by very many people. I believe this is because the defaults for some of the commands are unintuitive and don't match what people are used to using with [Bootmagic](https://docs.qmk.fm/#/feature_bootmagic?id=configuration).

This is primarily true for `MAGIC_KEY_BOOTLOADER`, which is the main reason why I'm proposing this change. Currently, the default Command for putting your board into bootloader mode is: **LShift+RShift+Pause**. This is silly because people rarely have Pause/Break in their keymaps. Instead, **LShift+RShift+B** would be a much better option, and it would match the **Space+B** combo used in Bootmagic. (LShift+RShift+Esc is another good option, so I've added that as an alternative.)

With that in mind, I propose the following changes to the Command keycodes' default values:

- Change `MAGIC_KEY_BOOTLOADER` default to `B` — to match Bootmagic's **Space+B**.
- Add `MAGIC_KEY_BOOTLOADER_ALT` and set it to `ESC` — to match [Bootmagic Lite](https://docs.qmk.fm/#/feature_bootmagic?id=bootmagic-lite)'s reset key (usually **Esc**).
- Add `MAGIC_KEY_EEPROM_CLEAR` and set it to `BSPACE` — to match Bootmagic's **Space+Backspace**.
- Remove `MAGIC_KEY_LAYER0_ALT2` — since `ESC` is now used as the bootloader alternate key.
- Rename `MAGIC_KEY_LAYER0_ALT1` to `MAGIC_KEY_LAYER0_ALT` — to be consistent with the other alternate key names.
- Rename `MAGIC_KEY_HELP1` to `MAGIC_KEY_HELP` and `MAGIC_KEY_HELP2` to `MAGIC_KEY_HELP_ALT` — to be consistent with the other alternate key names.

Thus the updated keycodes are:

|Constant name             |Default value|
|--------------------------|-------------|
|`MAGIC_KEY_BOOTLOADER`    |`B`          |
|`MAGIC_KEY_BOOTLOADER_ALT`|`ESC`        |
|`MAGIC_KEY_EEPROM_CLEAR`  |`BSPACE`     |
|`MAGIC_KEY_LAYER0_ALT`    |`GRAVE`      |
|`MAGIC_KEY_HELP`          |`H`          |
|`MAGIC_KEY_HELP_ALT`      |`SLASH`      |

I've also:

- updated templates that reference these keys;
- updated the Command feature docs;
- allowed the EEPROM config to be printed regardless of whether Bootmagic is turned on;
- removed some `MAGIC_KEY_*` defines from my and my friend's keymaps since they are now redundant.

I have **not** updated the plethora of commented out lines in keyboards' `config.h` files that use the old constant names and values. These are leftovers from the old AVR template, and seeing as they're just comments, I don't think they're worth updating (at least not right now). [Here's an example.](https://github.com/qmk/qmk_firmware/blob/76f744a75d636cba14b741bbe10f6a0fe5d58e04/keyboards/wilba_tech/wt65_a/config.h#L114-L139) To further prove this point, only two boards in all of QMK have any of these lines uncommented:

<pre><details>
<summary>rg -P '(?&lt;!//)#define MAGIC_KEY' -g '!tmk_core'</summary>
keyboards/atomic/config.h
131:#define MAGIC_KEY_BOOTLOADER       B

keyboards/atomic/keymaps/pvc/config.h
119:#define MAGIC_KEY_BOOTLOADER       B

keyboards/handwired/retro_refit/config.h
94:#define MAGIC_KEY_LOCK             BSLS
</details></pre>

## Types of changes
- [x] Core
- [ ] Bugfix
- [ ] New Feature
- [x] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [x] Documentation

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by vomindoraan** [comment]
If anyone's wondering, the only keyboard in all of QMK that redefines any of the `MAGIC_KEY_*` codes to use custom values is the Atomic — and that define is:

https://github.com/qmk/qmk_firmware/blob/b20704ecaf8b2dd043a0da6c3939b8d909001c94/keyboards/atomic/config.h#L131

which is exactly in line with this PR.

**Comment by vomindoraan** [comment]
Merged master and fixed conflict.

**Comment by vomindoraan** [comment]
@mechmerlin Could I get you to take a look at this, by any chance? 😃

**Comment by drashna** [comment]
Thanks!

**Comment by vomindoraan** [comment]
Cheers guys ❤️
